### ISSUE TITLE
 New issue HTTP Error 401

### ISSUE THREAD

**Issue by ghost** [calmh/unifi-api#36]
I am experiencing a HTTP Error 401. I am using the latest unifi-api and choose as version v4. 
My controller firmware is: v4.6.4

urllib2.HTTPError: HTTP Error 401: Unauthorized

Before i updated my Controller to 4.6.4 i was on 3.x and everything was working. After i updated my Controller i changed the version from v3 to v4 when instantiating the controller and ran in this problem

any idea ? Is version 4.6.4 already supported ?

**Comment by rapsta** [comment]
What meta->error, and what meta->msg is returned by the controller?

v4 returns a json-object on 401-error

**Comment by zFr3eak** [comment]
/usr/local/bin# python unifi-ls-clients -c UniFi -u admin -p 'xxxxxxxxx' -v v4 -s xxxxxxxxx
Traceback (most recent call last):
  File "unifi-ls-clients", line 17, in <module>
    aps = c.get_aps()
  File "/usr/local/lib/python2.7/dist-packages/unifi/controller.py", line 137, in get_aps
    return self._read(self.api_url + 'stat/device', params)
  File "/usr/local/lib/python2.7/dist-packages/unifi/controller.py", line 76, in _read
    res = self.opener.open(url, params)
  File "/usr/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 401: Unauthorized

**Comment by zFr3eak** [comment]
No one ?

**Comment by vitosans** [comment]
did you try changing your version to v4?

**Comment by oza4h07** [comment]
@vitosans:
I think he did change to v4: he installed  v4.6.4 and added -v v4 as argument to  unifi-ls-clients.
Maybe you were thinking of something else ?

**Comment by rjt** [comment]
On 20160209, using the `pip install -U unifi` version would still result in 401 UnAuthorized.

Figuring that the pip version has not been updated, i thought i would try the version directly from github.
Using the `git clone git@github.com:calmh/unifi-api.git` version, it worked.

So the code is good, but the pip package is still out-of-date.

**Comment by dilruacs** [comment]
pip install -U unifi results in 401 UnAuthorized for me too.

I have controller version 4.8.12.

"git clone https://github.com/calmh/unifi-api; cd unifi-api; python setup.py install", gets me an "Unauthorized" too

**Comment by dilruacs** [comment]
I solved my issue.

It probably isn't a problem of the API but a problem with the unifi controller.

I renamed the "default" site to a more descriptive name in the unifi web-UI

If I use the descriptive name in the API call, I get "401 Unauthorized", if I use "default", I get the output I expect.

Apparently the Unifi Controller just puts a different label on the "default" site if you rename it, but the API still has to use "default", will try to file a bug with the controller developpers.
